src/sinks/datadog/logs/sink.rs

```diff
@@ -136,6 +138,9 @@ impl Encoder<Vec<Event>> for DatadogLogsJsonEncoding {
             if let Some(Value::Timestamp(ts)) = log.remove(self.log_schema.timestamp_key()) {
                 log.insert_flat("timestamp", Value::Integer(ts.timestamp_millis()));
             }
+            if self.enterprise {
+                log.insert("source", "vector_enterprise");
```

Comment:
Setting it on `internal_logs` would avoid us overriding correctly set `.ddsource` when it's coming from `datadog_agent` or erroneously applying it to something that's not from Datadog Agents or us, like a `fluent` source or what not.

Reply:
I'm not sure I agree. Why would we set the `ddsource` field in the `internal_logs` source? Don't we only want to do that for this specific use case? IMO, we should just add a transform in the hidden topology since it's specific to that purpose. In the future this sounds like something schemas could solve (a semantic "source" field), but we're not there yet.

Reply:
> overriding correctly set .ddsource when it's coming from datadog_agent or erroneously applying it to something that's not from Datadog Agents or us, like a fluent source or what not.

A transform that only affects the `internal_logs` source is also fine - and is the eventual semantic meanings, assuming the contents of the semantic field matches what we expect in Datadog. For example, if `source_type` semantically mapped to `ddsource`, having the value be `internal_logs` doesn't help.